cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -151,18 +152,17 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
   @Override
   public void setProperties(final MetadataScope scope, final MetadataEntity metadataEntity,
                             final Map<String, String> currentProperties) {
-    final AtomicReference<MetadataRecordV2> previousRef = new AtomicReference<>();
-    execute(mds -> {
-      MetadataDataset.MetadataChange metadataChange = mds.setProperty(metadataEntity, currentProperties);
-      previousRef.set(new MetadataRecordV2(metadataEntity, scope, metadataChange.getExisting().getProperties(),
-                                           metadataChange.getExisting().getTags()));
+    MetadataChange metadataChange = execute(mds -> {
+      return mds.setProperty(metadataEntity, currentProperties);
```

Comment:
When it's a single line, I don't think you need the brackets or the return either, can just do:
```
execute(mds -> mds.setProperty(metadataEntity, currentProperties), scope);
```

Reply:
This does not seem to work in this case, I think because the function is being passed as a para and the compiler cannot unambiguously determine which one I am calling without specifying return.